image_transport/src/publisher.cpp

```diff
@@ -110,9 +110,23 @@ Publisher::Publisher(
   impl_->base_topic_ = image_topic;
   impl_->loader_ = loader;
 
+  uint ns_len = node->get_effective_namespace().length();
```

Comment:
Can ``param_base_name`` be just equal to ``base_topic``, rather than stripping off the namespace here?

Reply:
@gbiggs @ijnek This is comes from `image_transport_plugins`.
Which is better?

- use base topic name
  This looks like below.
 ```
/usb_cam:
  ros__parameters:
    image_raw:
      enable_pub_plugins:
      - image_transport/compressed
      format: jpeg
      jpeg_quality: 95
      png_level: 3
      tiff:
        res_unit: inch
        xdpi: -1
        ydpi: -1
```
- use whole topic name
  This looks like below.
```
/usb_cam:
  ros__parameters:
    image_raw:
      format: jpeg
      jpeg_quality: 95
      png_level: 3
      tiff:
        res_unit: inch
        xdpi: -1
        ydpi: -1
    /aaa/bbb/image_raw:
       enable_pub_plugins:
       - image_transport/compressed
```

Reply:
Just the base topic name seems better.